src/plugins/intel_gpu/src/graph/impls/onednn/fully_connected_onednn.cpp

```diff
@@ -303,6 +303,7 @@ struct fully_connected_onednn : typed_primitive_onednn_impl<fully_connected> {
                     // 8-bit quantized weight
                     attr->set_scales(DNNL_ARG_WEIGHTS, PER_OC, dnnl::memory::dims{}, ds_data_type);
                 } else {
+                    OPENVINO_ASSERT((ngroups == 1 || group_size % 32 == 0), "group_size should be aligned to 32 if it is not a single scale group");
```

Comment:
1. The group size % 32 is not only for 4 bit weight. It is a limitation regarding performance from onednn perspective. Lets move this above. 
2. also even though the scale group size is 32, for int4 +  K shape unaligned to 32 (e.g., K is 62 ), it is just an onednn API limitation, we can request this support in the future if the data is padded as 64.

Reply:
Move it above condition